### PR TITLE
[Doc] add 3.3 release notes

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by EsoragotoSpirit — DISMISSED** [review]

File: docs/zh/release_notes/release-3.3.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
+displayed_sidebar: "Chinese"
+---
+
+# StarRocks version 3.2
```
Comment (EsoragotoSpirit) [inline docs/zh/release_notes/release-3.3.md:5]: 3.2？

File: docs/zh/release_notes/release-3.3.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+## 3.2.6
```
Comment (EsoragotoSpirit) [inline docs/zh/release_notes/release-3.3.md:7]: 3.2.6？

File: docs/en/release_notes/release-3.3.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+Fixed the following issues:
+
+- Query results are incorrect when queries are rewritten to materialized views created by using UNION ALL. [#42949](https://github.com/StarRocks/starrocks/issues/42949)
+- If  BEs are compiled with ASAN, BEs crash when the cluster is used and the be.warning log shows `dict_func_expr == nullptr`. [#44551](https://github.com/StarRocks/starrocks/issues/44551)
+- Query results are incorrect when aggregate queries are performed on single-replica tables. [#43223](https://github.com/StarRocks/starrocks/issues/43223)
+- View Delta Join rewriting fails. [#43788](https://github.com/StarRocks/starrocks/issues/43788)
+- BEs crash after the type of a column is modified from VARCHAR to DECIMAL. [#44406](https://github.com/StarRocks/starrocks/issues/44406)
+- When a table with List partitioning is queried by using a not equal operator, partitions are incorrectly pruned, resulting in wrong query results. [#42907](https://github.com/StarRocks/starrocks/issues/42907)
+- Leader FE's heap size increases quickly as many Stream Load jobs with non-transactional  interface finishes. [#43715](https://github.com/StarRocks/starrocks/issues/43715)
```
Comment (EsoragotoSpirit) [inline docs/en/release_notes/release-3.3.md:99]: extra space before interface

File: docs/en/release_notes/release-3.3.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+- **Optimized global dictionary:** provides a dictionary object to store the mapping of key-value pairs from a dictionary table in the memory of each BE node. By using the `dictionary_get()` function to directly query the dictionary object in the BE memory, the speed is faster compared to using the `dict_mapping` function to query the dictionary table. Furthermore, the dictionary object can also serve as a dimension table. Dimension values can be obtained by directly querying the dictionary object through the `dictionary_get()` function, resulting in faster query speeds compared to the original method of performing JOIN operations on the dimension table to obtain dimension values.
+- Supports Colocate Group Execution: significantly reduced memory usage for conducting Join and Agg operators on the colocated tables, which ensures that large queries can be executed more stably.
+- Optimized the performance of CodeGen: JIT is enabled by default, which can achieve a 5X performance improvement for complex expression calculations.
+- Supports using vectorization technology to implement regular expressions matching, which can reduce the CPU consumption of the `regexp_replace` function.
```
Comment (EsoragotoSpirit) [inline docs/en/release_notes/release-3.3.md:46]: regular expressions matching

删除s

File: docs/en/release_notes/release-3.3.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+- When an aggregate query contains `Limit`, multiple Pipeline threads can share the `Limit` condition to prevent wasting computational resources.
+
+### Storage Optimization and Cluster Management
+
+- **[Enhanced flexibility of range partitioning](https://docs.starrocks.io/docs/3.3/table_design/Data_distribution/#range-partitioning):** three specific time functions can be used as partitioning columns, so that these functions can convert timestamps or strings in the  partitioning columns into date values and then the data can be partitioned based on the converted date values.
+- **FE memory observability**: Provides detailed memory usage metrics for each module within the FE to better manage resources.
+- **[Optimized metadata locks in FE](https://docs.starrocks.io/zh/docs/3.3/administration/management/FE_configuration/#lock_manager_enabled)**: provides Lock manager to perform centralized management for metadata locks in FE. For example, it can refine the granularity of metadata lock from the database level to the table level, which can improve load and queries concurrency. In a scenario of 100 concurrent data loading jobs, loading time can be reduced by 35%.
```
Comment (EsoragotoSpirit) [inline docs/en/release_notes/release-3.3.md:55]: queries concurrency --> query concurrency

File: docs/en/release_notes/release-3.3.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  - Array functions: [array_contains_seq](https://docs.starrocks.io/docs/3.3/sql-reference/sql-functions/array-functions/array_contains_seq/)
+  - Date and time functions: [yearweek](https://docs.starrocks.io/docs/3.3/sql-reference/sql-functions/date-time-functions/yearweek/)
+  - Math functions: [crbt](https://docs.starrocks.io/docs/3.3/sql-reference/sql-functions/math-functions/cbrt/)
+
+### Ecosystem Support
+
+- Provides [ClickHouse SQL Rewriter](https://github.com/StarRocks/SQLTransformer), a new tool for converting the syntax in Clickhouse to the syntax in StarRocks.
+- The Flink connector v1.2.9 provided by StarRocks is integrated with the Flink CDC 3.0 framework, which can construct a streaming ELT pipeline from CDC data sources to StarRocks. The pipeline can synchronize whole database, merged sharding tables, and schema changes from sources to StarRocks. For more information,  see [Synchronize data with Flink CDC 3.0 (with schema change supported)](https://docs.starrocks.io/docs/loading/Flink-connector-starrocks/#synchronize-data-with-flink-cdc-30-with-schema-change-supported).
```
Comment (EsoragotoSpirit) [inline docs/en/release_notes/release-3.3.md:87]: whole database --> a database as a whole?

merged sharding tables --> merge

File: docs/zh/release_notes/release-3.3.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+### 存储优化与集群管理
+
+- **[增强 Range 分区的灵活性](https://docs.starrocks.io/zh/docs/3.3/table_design/Data_distribution/#range-分区)**：新增支持三个特定时间函数作为分区列，可以将时间戳或字符串的分区列值转成日期，然后按转换后的日期划分分区。
+- **FE 内存可观测性**：提供 FE 内各模块的详细内存使用指标，以便更好地管理资源。
+- **[优化 FE 中的元数据锁](https://docs.starrocks.io/zh/docs/3.3/administration/management/FE_configuration/#lock_manager_enabled)**：提供 Lock manager，可以对 FE  中的元数据锁实现集中管理，例如将元数据锁的粒度从库级别细化为表级别，可以提高导入和查询的并发性能。在 100 并发的导入场景下，导入耗时减少 35%。
+- **[使用标签管理 BE](https://docs.starrocks.io/zh/docs/3.3/administration/management/resource_management/be_label/)**：支持基于 BE 节点所在机架、数据中心等信息，使用标签对 BE 节点进行分组，以保证数据在机架或数据中心等之间均匀分布，应对某些机架断电或数据中心故障情况下的灾备需求。
+- **[优化排序键](https://docs.starrocks.io/zh/docs/3.3/table_design/indexes/Prefix_index_sort_key/)**：明细表、聚合表和更新表均支持通过 `ORDER BY` 子句指定排序键。
+- **优化非字符串标量类型数据的存储效率**：这类数据支持字典编码，存储空间下降 12%。
+- **主键表支持 Size-tiered compactio 策略**：降低执行 compaction 时写 IO 和内存开销。存算分离和存算一体集群均支持该优化。
+- **优化主键表持久化索引的读 IO**：支持按照更小的粒度（页）读取持久化索引，并且改进持久化索引的 bloom filter。存算分离和存算一体集群均支持该优化。
```
Comment (EsoragotoSpirit) [inline docs/zh/release_notes/release-3.3.md:66]: I/O

File: docs/zh/release_notes/release-3.3.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+### 存储优化与集群管理
+
+- **[增强 Range 分区的灵活性](https://docs.starrocks.io/zh/docs/3.3/table_design/Data_distribution/#range-分区)**：新增支持三个特定时间函数作为分区列，可以将时间戳或字符串的分区列值转成日期，然后按转换后的日期划分分区。
+- **FE 内存可观测性**：提供 FE 内各模块的详细内存使用指标，以便更好地管理资源。
+- **[优化 FE 中的元数据锁](https://docs.starrocks.io/zh/docs/3.3/administration/management/FE_configuration/#lock_manager_enabled)**：提供 Lock manager，可以对 FE  中的元数据锁实现集中管理，例如将元数据锁的粒度从库级别细化为表级别，可以提高导入和查询的并发性能。在 100 并发的导入场景下，导入耗时减少 35%。
+- **[使用标签管理 BE](https://docs.starrocks.io/zh/docs/3.3/administration/management/resource_management/be_label/)**：支持基于 BE 节点所在机架、数据中心等信息，使用标签对 BE 节点进行分组，以保证数据在机架或数据中心等之间均匀分布，应对某些机架断电或数据中心故障情况下的灾备需求。
+- **[优化排序键](https://docs.starrocks.io/zh/docs/3.3/table_design/indexes/Prefix_index_sort_key/)**：明细表、聚合表和更新表均支持通过 `ORDER BY` 子句指定排序键。
+- **优化非字符串标量类型数据的存储效率**：这类数据支持字典编码，存储空间下降 12%。
+- **主键表支持 Size-tiered compactio 策略**：降低执行 compaction 时写 IO 和内存开销。存算分离和存算一体集群均支持该优化。
```
Comment (EsoragotoSpirit) [inline docs/zh/release_notes/release-3.3.md:65]: compactio --> compaction

File: docs/zh/release_notes/release-3.3.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+### 数据湖分析
+
+- **Data Cache 增强**：
+  - 新增 [缓存预热 (Warmup)](https://docs.starrocks.io/zh/docs/3.3/data_source/data_cache_warmup/) 命令 CACHE SELECT，用于填充查询热点数据，可以结合 SUBMIT TASK 完成周期性填充。
+  - 增加了多项 [Data Cache 可观测性指标](https://docs.starrocks.io/zh/docs/3.3/data_source/data_cache_observe/)。
+- **Parquet Reader 性能提升**：
+  - 针对 Page Index 的优化，显著减少 Scan 数据规模。
+  - 在有 Page Index 的情况下，降低 Page 多读的情况。
+  - 使用 simd 加速计算判断数据行是否为空。
+- **ORC Reader性能提升**：
```
Comment (EsoragotoSpirit) [inline docs/zh/release_notes/release-3.3.md:28]: Reader 后空格

File: docs/zh/release_notes/release-3.3.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+### 性能提升和查询优化
+
+- **中间结果落盘（Spill to Disk）能力 GA**：优化复杂查询的内存占用，优化 Spill 的调度，保证大查询都能稳定执行，不会 OOM。
+- **支持更多索引**：
+  - 支持[全文倒排索引](https://docs.starrocks.io/zh/docs/3.3/table_design/indexes/inverted_index/)，以加速全文检索。
+  - 支持 [N-Gram bloom filter 索引](https://docs.starrocks.io/zh/docs/3.3/table_design/indexes/Ngram_Bloom_Filter_Index/)，以加速 `LIKE` 查询或 `ngram_search` 和 `ngram_search_case_insensitive` 函数的运算速度。
+- 提升 Bitmap 系列函数的性能并优化内存占用，补充了 Bitmap 导出到 Hive 的能力以及配套的 [Hive Bitmap UDF](https://docs.starrocks.io/zh/docs/3.3/integrations/hive_bitmap_udf/)。
+- **支持 [Flat JSON](https://docs.starrocks.io/zh/docs/3.3/using_starrocks/Flat_json/)**：导入时自动检测 JSON 数据并提取公共字段，自动创建 JSON 的列式存储结构，让 JSON 查询性能达到 Struct 的水平。
+- **优化全局字典**：提供字典对象，将字典表中的键值对映射关系存在各个 BE 节点的内存中。通过 `dictionary_get()` 函数直接查询 BE 内存中的字典对象，相对于原先使用 `dict_mapping` 函数查询字典表，查询速度更快。并且字典对象还可以作为维度表，可以通过 `dictionary_get()` 函数直接查询字典对象来获取维度值，相对于原先通过 JOIN 维度表来获取维度值，查询速度更快。
+- 支持 Colocate Group Execution:  大幅降低在 colocated 表上执行 Join 和 Agg 算子时的内存占用，能够更稳定的执行大查询。
```
Comment (EsoragotoSpirit) [inline docs/zh/release_notes/release-3.3.md]: 冒号后两个空格

File: docs/en/release_notes/release-3.3.md (hunk 6 above)
Comment (EsoragotoSpirit) [reply docs/en/release_notes/release-3.3.md:87]: extra space before see

File: docs/en/release_notes/release-3.3.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+### Bug Fixes
+
+Fixed the following issues:
+
+- Query results are incorrect when queries are rewritten to materialized views created by using UNION ALL. [#42949](https://github.com/StarRocks/starrocks/issues/42949)
+- If  BEs are compiled with ASAN, BEs crash when the cluster is used and the be.warning log shows `dict_func_expr == nullptr`. [#44551](https://github.com/StarRocks/starrocks/issues/44551)
```
Comment (EsoragotoSpirit) [inline docs/en/release_notes/release-3.3.md:94]: extra space ahead of BE

File: docs/zh/release_notes/release-3.3.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+
+### 存储优化与集群管理
+
+- **[增强 Range 分区的灵活性](https://docs.starrocks.io/zh/docs/3.3/table_design/Data_distribution/#range-分区)**：新增支持三个特定时间函数作为分区列，可以将时间戳或字符串的分区列值转成日期，然后按转换后的日期划分分区。
+- **FE 内存可观测性**：提供 FE 内各模块的详细内存使用指标，以便更好地管理资源。
+- **[优化 FE 中的元数据锁](https://docs.starrocks.io/zh/docs/3.3/administration/management/FE_configuration/#lock_manager_enabled)**：提供 Lock manager，可以对 FE 中的元数据锁实现集中管理，例如将元数据锁的粒度从库级别细化为表级别，可以提高导入和查询的并发性能。在 100 并发的导入场景下，导入耗时减少 35%。
+- **[使用标签管理 BE](https://docs.starrocks.io/zh/docs/3.3/administration/management/resource_management/be_label/)**：支持基于 BE 节点所在机架、数据中心等信息，使用标签对 BE 节点进行分组，以保证数据在机架或数据中心等之间均匀分布，应对某些机架断电或数据中心故障情况下的灾备需求。
+- **[优化排序键](https://docs.starrocks.io/zh/docs/3.3/table_design/indexes/Prefix_index_sort_key/)**：明细表、聚合表和更新表均支持通过 `ORDER BY` 子句指定排序键。
+- **优化非字符串标量类型数据的存储效率**：这类数据支持字典编码，存储空间下降 12%。
+- **主键表支持 Size-tiered compactio 策略**：降低执行 compaction 时写 I/O 和内存开销。存算分离和存算一体集群均支持该优化。
```
Comment (EsoragotoSpirit) [inline docs/zh/release_notes/release-3.3.md:65]: compactio --> compaction

File: docs/en/release_notes/release-3.3.md [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+- **[Supports adding labels on BEs](https://docs.starrocks.io/docs/3.3/administration/management/resource_management/be_label/):** supports adding labels on BEs based on information such as the racks and data centers where BEs are located. It ensures that data can be evenly distributed among racks, data centers or others, to address disaster recovery requirements in case that certain racks lose power or data centers encounter failures.
+- **[Optimized the sort key](https://docs.starrocks.io/docs/3.3/table_design/indexes/Prefix_index_sort_key/#usage-notes):** Duplicate Key tables, Aggregate tables, and Unique Key tables all support specifying sort keys through the `ORDER BY` clause.
+- **Optimized the storage efficiency of non-string scalar data:** This type of data supports dictionary encoding, which reduces storage space by 12%.
+- **Supports the Size-tiered compaction strategy for primary key tables:** reduced write IO and memory overhead during compaction. This improvement is supported in both shared-data and shared-nothing StarRocks clusters.
```
Comment (EsoragotoSpirit) [inline docs/en/release_notes/release-3.3.md:59]: IO --> I/O

**Review by evelynzhaojie — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Pull request merged**